### PR TITLE
User system upgrade fixes

### PR DESCRIPTION
**Pull request by jyejare**
This PR fixes the Issues in Upgrade automation for below issues:
1. #450 - Fix issues Upgrading Satellite/Capsule to CDN version
2. #449 - Fix issues Upgrading User Satellite and Capsule
3. Some functions added in hammer library

### CODE REVIEW

File: automation_tools/satellite6/hammer.py [hunk 1]
```diff
@@ -299,6 +317,27 @@ def hammer_capsule_list():
     return hammer('capsule list')
 
 
+@task
+def hammer_activation_key_contnet_override(
```
Comment (sghai) [inline automation_tools/satellite6/hammer.py]: `hammer_activation_key_content_override`

File: automation_tools/satellite6/upgrade/tasks.py [hunk 2]
```diff
@@ -13,11 +13,13 @@
     get_product_subscription_id,
     hammer,
     hammer_activation_key_add_subscription,
+    hammer_activation_key_contnet_override,
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/tasks.py]: `hammer_activation_key_content_override`

File: automation_tools/satellite6/upgrade/tasks.py [hunk 3]
```diff
@@ -361,9 +363,16 @@ def sync_capsule_repos_to_upgrade(capsules):
 
     CAPSULE_URL
         The url for capsule repo from latest satellite compose.
+        Optional, the TO_VERSION Caspule repo from Redhat Repos will be used
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/tasks.py]: It should be rephrased like: If not provided, capsule repo from Red Hat repositories would be enabled.

File: automation_tools/satellite6/upgrade/tasks.py [hunk 4]
```diff
@@ -398,17 +406,27 @@ def sync_capsule_repos_to_upgrade(capsules):
...
     # Create product capsule
-    hammer_product_create('capsule6_latest', '1')
-    time.sleep(2)
-    hammer_repository_create(
-        'capsule6_latest_repo', '1', 'capsule6_latest', capsule_repo)
-    hammer_repository_synchronize(
-        'capsule6_latest_repo', '1', 'capsule6_latest')
+    if capsule_repo:
+        hammer_product_create(product_name, '1')
+        time.sleep(2)
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/tasks.py]: why we need sleep here ?

File: automation_tools/satellite6/upgrade/tasks.py [hunk 5]
```diff
@@ -421,21 +439,30 @@ def sync_capsule_repos_to_upgrade(capsules):
         '{} '.format(cv_name))[1]) for data in cv_version_data]).pop()
     cv_ver_id = get_attribute_value(cv_version_data, '{0} {1}'.format(
         cv_name, latest_cv_ver), 'id')
-    hammer_content_view_promote_version(cv_name, cv_ver_id, lc_env_id, '1')
-    # Add new product subscriptions to AK
-    hammer_activation_key_add_subscription(ak_name, '1', 'capsule6_latest')
+    hammer_content_view_promote_version(
+        cv_name, cv_ver_id, lc_env_id, '1')
+    # If Downstream, Add new product subscriptions to AK
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/tasks.py]: In case custom capsule repo URL given, update AK with capsule product subscription.

**Comment by sghai** [comment]
ACK pending few comments.

File: automation_tools/satellite6/upgrade/capsule.py [hunk 6]
```diff
@@ -58,9 +58,13 @@ def satellite6_capsule_setup(sat_host, os_version):
...
         execute(lambda: run('katello-service restart'), host=cap_hosts)
     env['capsule_hosts'] = cap_hosts
+    if ',' in cap_hosts:
+        cap_hosts = cap_hosts.split(',')
+    else:
+        cap_hosts = [cap_hosts]
     copy_ssh_key(sat_host, cap_hosts)
-    if os.environ.get('CAPSULE_URL') is not None:
-        execute(sync_capsule_repos_to_upgrade, cap_hosts, host=sat_host)
+    execute(sync_capsule_repos_to_upgrade, cap_hosts, host=sat_host)
```
Comment (jyejare) [inline automation_tools/satellite6/upgrade/capsule.py]: No, Both Syncing of capsule repos and tools repos has been separated in separate functions.
Now, there are :
1. sync_capsule_repos_to_upgrade
2. sync_tools_repos_to_upgrade
As per need, User can choose to go for Capsule Upgrade / Client Upgrade, both are not combined.

File: automation_tools/satellite6/upgrade/tasks.py (hunk 4 above)
Comment (jyejare) [inline automation_tools/satellite6/upgrade/tasks.py]: To avoid any race condition !

**Comment by jyejare** [comment]
@sghai , I worked on your comments. Please revisit to merge/comments if any.

**Comment by jyejare** [comment]
ping @sghai , @elyezer !!

**Comment by sghai** [comment]
ACK.. looks good. Merging now.

**Pull request merged**